File: src/main/java/j/Coach.java
package j;

import k.Seat;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

@Deprecated
public final class Coach implements k.RailroadCar {

    private final List<Seat> seats;

    public Coach(final int availableSeats) {
        if (availableSeats < 1) {
            throw new IllegalArgumentException("availableSeats " + availableSeats);
        }
        seats = IntStream.range(0, availableSeats)
                .mapToObj((i) -> new Seat())
                .collect(Collectors.toList());
    }

    @Override
    public int getWeight() {
        return 5;
    }

    @Override
    public String toString() {
        return "[" +
                seats.stream()
                        .map(Object::toString)
                        .collect(Collectors.joining(""))
                + "]";
    }
}

File: src/main/kotlin/k/Coach.kt
package k

class Coach(availableSeats: Int) : k.RailroadCar {

    init {
        require(availableSeats > 0) { "availableSeats $availableSeats" }
    }

    val seats: List<Seat> = (1..availableSeats).map { Seat() }

    override val weight: Int
        get() = 5

    override fun toString(): String =
            seats.joinToString(separator = "", prefix = "[", postfix = "]")
}
